### DISCUSSION TITLE
I can't save preferences because I can't connect to https://localhost

### DISCUSSION THREAD

**Discussion by ghost** [searxng/searxng#1491] [Q&A] [answered]
Hi, I have just set up a private instance with Docker on Arch Linux. However, it only starts with http://localhost:8080 and when I try to save the preferences I get an "unable to connect to https://localhost" error. What's wrong? Thanks in advance.

**Comment by mrpaulblack** [comment, marked as answer]
Hi,
I think you did not set your base URL correctly... Make sure to change the SEARXNG_HOSTNAME in your .env file to http://localhost:8080 and to set your BASE_URL for SearXNG in searxng/settings.yml here: https://github.com/searxng/searxng/blob/b6d59decda2be89a9b96e29b7a12538ba4af2c5c/searx/settings.yml#L58

**Reply by ghost** [reply]
Hi Paul, I only have a settings.yml in the /usr/local /Searxng-docker directory and it doesn't have the line that you mentioned, am I looking for in the wrong place?

**Reply by mrpaulblack** [reply]
Hi if the `settings.yml` file contains:
```use_default_settings: true``` it will use default settings and then you can add on custom changes in that file...

Just add the `base_url: "http://localhost:8080"` after https://github.com/searxng/searxng-docker/blob/master/searxng/settings.yml#L3= (make sure the indentation is correct) :+1:

**Reply by ghost** [reply]
Done, but now I get a server error and my bar has https://http//localhost:8080😳😳

**Reply by mrpaulblack** [reply]
hmmm maybe it has a problem with http? @dalf Do you know where the BASE_URL var is used to produce the links for preference safe for example?

**Reply by ghost** [reply]
Maybe in docker-compose.yaml?
UPDATE: yess it was there, I just had to remove https:// before the variable, now it works, thanks a lot! 😊

**Reply by mrpaulblack** [reply]
oh ok, I will mark this as solved; glad you could resolve it :D
